drivers/sensor/wsen_tids/wsen_tids.c

```diff
@@ -0,0 +1,228 @@
+static const int32_t tids_odr_list[] = {
+	25,
+	50,
+	100,
+	200,
+};
+
+static int tids_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	struct tids_data *data = dev->data;
+	int16_t raw_temperature;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL);
+
+	if (TIDS_getRawTemperature(&data->sensor_interface, &raw_temperature) != WE_SUCCESS) {
+		LOG_ERR("Failed to fetch data sample");
+		return -EIO;
+	}
+
+	data->temperature = raw_temperature;
+
+	return 0;
+}
+
+static int tids_channel_get(const struct device *dev, enum sensor_channel chan,
+			    struct sensor_value *val)
+{
+	struct tids_data *data = dev->data;
+
+	if (chan == SENSOR_CHAN_AMBIENT_TEMP) {
+		/* Convert temperature from 0.01 degrees Celsius to degrees Celsius */
+		val->val1 = data->temperature / 100;
+		val->val2 = ((int32_t)data->temperature % 100) * (1000000 / 100);
+	} else {
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+/* Set output data rate. See tids_odr_list for allowed values. */
+static int tids_odr_set(const struct device *dev, const struct sensor_value *odr)
+{
+	struct tids_data *data = dev->data;
+	int odr_index;
+
+	for (odr_index = 0; odr_index < ARRAY_SIZE(tids_odr_list); odr_index++) {
+		if (odr->val1 == tids_odr_list[odr_index] && odr->val2 == 0) {
+			break;
+		}
+	}
+
+	if (odr_index == ARRAY_SIZE(tids_odr_list)) {
+		/* ODR not allowed (was not found in tids_odr_list) */
+		LOG_ERR("Bad sampling frequency %d.%d", odr->val1, abs(odr->val2));
+		return -EINVAL;
+	}
+
+	if (TIDS_setOutputDataRate(&data->sensor_interface, (TIDS_outputDataRate_t)odr_index) !=
+	    WE_SUCCESS) {
+		LOG_ERR("Failed to set output data rate");
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int tids_attr_set(const struct device *dev, enum sensor_channel chan,
+			 enum sensor_attribute attr, const struct sensor_value *val)
+{
+	if (chan != SENSOR_CHAN_ALL) {
+		LOG_WRN("attr_set() is not supported on channel %d.", chan);
+		return -ENOTSUP;
+	}
+
+	switch (attr) {
+	case SENSOR_ATTR_SAMPLING_FREQUENCY:
+		return tids_odr_set(dev, val);
+
+#ifdef CONFIG_WSEN_TIDS_TRIGGER
+	case SENSOR_ATTR_LOWER_THRESH:
+		return tids_threshold_set(dev, val, false);
+
+	case SENSOR_ATTR_UPPER_THRESH:
+		return tids_threshold_set(dev, val, true);
+#endif /* CONFIG_WSEN_TIDS_TRIGGER */
+
+	default:
+		LOG_ERR("Operation not supported.");
+		return -ENOTSUP;
+	}
+}
+
+static const struct sensor_driver_api tids_driver_api = {
+	.attr_set = tids_attr_set,
+#if CONFIG_WSEN_TIDS_TRIGGER
+	.trigger_set = tids_trigger_set,
+#endif
+	.sample_fetch = tids_sample_fetch,
+	.channel_get = tids_channel_get,
+};
+
+static int tids_init(const struct device *dev)
+{
+	const struct tids_config *const config = dev->config;
+	struct tids_data *data = dev->data;
+	int status;
+	uint8_t device_id;
+	struct sensor_value odr;
+
+	/* Initialize WE sensor interface */
+	TIDS_getDefaultInterface(&data->sensor_interface);
+	data->sensor_interface.interfaceType = WE_i2c;
+#if DT_ANY_INST_ON_BUS_STATUS_OKAY(i2c)
```

Comment:
Is this sensor capable of supporting anything other than I2C? If not, then I'd remove the conditionals